data/src/main/java/com/microsoft/azure/kusto/data/StringUtils.java

```diff
@@ -13,6 +17,7 @@ public static String GetStringTail(String val, int minRuleLength) {
         return val.substring(val.length() - minRuleLength);
     }
 
-    private StringUtils() {
+    public static String quoteEntityName(String name) {
+        return "['" + name + "']";
```

Comment:
What if the table is already quoted? 
Because we don't need to escape chars, we can assume that if it has the prefix/suffix we don't need to add it:
`StringUtils.appendIfMissing(StringUtils.prependIfMissing(str, "['"), "']");`

Reply:
in spark we have this code and i really prefer this logic:
def normalizeTableName(table: String): String = {
    val tableName = table.replace("-", "_")

    if (tableName.startsWith("[")) {
      tableName
    }
    else if (!tableName.contains("'")) {
      "['" + tableName + "']"
    }
    else {
      "[\"" + tableName + "\"]"
    }
  }

Reply:
I think the intent with the quickstart is that they provide just the table name, and the documentation illustrates that as well. But I do agree, because I really intended this to be expanded for use outside quickstart as well, so I'll add some more logic to cover cases where it's already escaped.